### PR TITLE
Feature/multi session logout

### PR DESCRIPTION
**Pull request by etelmon**
- Allow multisession access (multiple tabs of a same browser instance)
- Reinforce token security to prevent use of refreshed token to maintain other user socket connection.
- Safer logout (log out all browser sessions at once if coming from same browser instance)
- Safer login (Make sure all active sessions of a same browser instance are on the same user)

### CODE REVIEW

File: lib/authorize.js [hunk 1]
```diff
@@ -58,49 +63,121 @@ function socketAuthorize(options, onConnection) {
...
+     * 
+     * @param {*} decodedToken could be a decoded access code or a token
+     * @param {*} context 
+     */
+    function initNewConnection(decodedToken, context) {
+        const currentSocket = context.socket;
+        currentSocket.userId = decodedToken.id;
+
+        if (context.data.origin ) {
+            invalidateAllSocketsAtThisOriginWhichAreNotRelatedToUser(currentSocket.server.sockets.sockets, context.data.origin, currentSocket.userId);
```
Comment (Jacob-Friesen) [inline lib/authorize.js]: This name is too long.

File: lib/authorize.js [hunk 2]
```diff
@@ -58,49 +63,121 @@ function socketAuthorize(options, onConnection) {
...
-    function authorizeWithAuthCodeAndRefreshedToken(decoded, onSuccess, onError, context) {
+    /**
+     * This function is called each time the client attempts to authenticate on the currently open socket.
+     * 
+     * @param {*} decodedToken 
+     * @param {*} onSuccess 
+     * @param {*} onError 
+     * @param {*} context 
+     */
+    function authorizeWithAuthCodeAndRefreshedToken(decodedToken, onSuccess, onError, context) {
```
Comment (Jacob-Friesen) [inline lib/authorize.js]: These need descriptions and types.

File: lib/authorize.js (hunk 1 above)
Comment (etelmon) [reply lib/authorize.js]: The idea is that when your function name is meaningful, there is no need for documentation.

The idea is the documentation is the code and we should have minimum number of comments in the code.

File: lib/authorize.js (hunk 2 above)
Comment (etelmon) [reply lib/authorize.js]: I ll do.

**Pull request merged**